onnxruntime/core/optimizer/graph_transformer.cc

```diff
@@ -7,11 +7,12 @@ using namespace ::onnxruntime::common;
 
 namespace onnxruntime {
 
-Status GraphTransformer::Apply(Graph& graph, bool& modified, const std::vector<std::string>& ) const {
+Status GraphTransformer::Apply(Graph& graph, bool& modified, 
+                               const std::vector<std::string>& compatible_provider_types) const {
   // the Graph should be in a good state prior this being called, so there should be no need to call Resolve here
```

Comment:
I would have expected the compatible provider types to be a property of the transformer, so wondering why it's a parameter to Apply

Reply:
I can change this to be a property in a heart beat... actually I had it as a property in my very first PR.... we decided that providers should not be a property of transformers instead it should be managed by transformer manager... during apply transformer manager can give a list of compatible providers...

I think making it a property will simplify the code... I spoke to Konstantinos and he also thinks it will simplify things... will talk to you offline in detail and we can decide...